### DISCUSSION TITLE
How to install python 3.8

### DISCUSSION THREAD

**Discussion by monkeycc** [termux/termux-app#3189] [General]
How to install python 3.8

```
pkg uninstall python -y
apt install python3.8 -y

pkg install tur-repo
pkg update
pkg install python3.8
```

```
python -V
bash: /data/data/com.termux/files/usr/bin/python: No such file or directory
```

**Comment by 2096779623** [comment]
`ln -s $PREFIX/bin/python3.8 $PREFIX/bin/python`

**Reply by monkeycc** [reply]
thank you
Problem solving

cd /data/data/com.termux/files/usr/bin
ln -sf python3.8 python

**Reply by Mojtaba2740** [reply]
I like to install library of python on termux but I do not success for error in locatated

**Comment by Preetham1729** [comment]
````````
   ~ apt install python3.8 -y
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package python3.8
E: Couldn't find any package by glob 'python3.8'

````````
What's the issue with my version of termux? Any solutions for this?

**Reply by TomJo2000** [reply]
You can try the workaround mentioned above;
```bash
ln -s $PREFIX/bin/python3.8 $PREFIX/bin/python
```
Which will symlink the current version of Python (3.11.5 as of writing), to python3.8
Thus typing in `python3.8` will run the latest version of python.
Which is probably going to be fine for whatever you want to run.

---

If it is not fine for whatever reason, and ***please test with the current version first***, then you can install [Python 3.8 from the TUR](https://github.com/termux-user-repository/tur/tree/master/tur/python3.8).
```bash
apt install tur-repo
apt install python3.8
```
Python version and environment management is a pain in the ass.
So if you can just use the current version, use it.

**Comment by Blgt11147** [comment]
python3: can't open file '/data/data/com.termux/files/home/ALL-IN-ONE/ALL.py': [Errno 2] No such file or directory

**Comment by ItsCitronus** [comment]
I installed Python 3.8, I type pip install in the terminal and it says The program pip is not installed. Install it by running
 pkg install python-pip.

When I install pip, instead of python 3.8 it says python 3.11.9, how can I fix this?

**Reply by optopi-codes** [reply]
cd to termux bin and just like for python do the same for pip
`
ln -sf pip3.8 pip
`

**Reply by twaik** [reply]
`ln -sf` command will not magically turn python 3.11.9 into 3.8...

**Reply by optopi-codes** [reply]
> `ln -sf` command will not magically turn python 3.11.9 into 3.8...

@ItsCitronus has installed python 3.8 but then doing "pkg install python-pip" will default install both latest python and pip (3.11)

When python3.8 is installed you are installing pip3.8 also, so instead of downloading pip via pkg doing ln -sf will work!
